tools/src/main/scala/com/nvidia/spark/rapids/tool/profiling/AutoTuner.scala

```diff
@@ -339,157 +572,226 @@ class AutoTuner(app: ApplicationSummaryInfo, workerInfo: String) extends Logging
-    recommendSparkProperties(recommendedConfig, systemProps)
-    recommendGpuProperties(recommendedConfig, systemProps)
-    (toPropertyProfileResult(recommendedConfig), toCommentProfileResult(comments))
+    (toRecommendationsProfileResult, toCommentProfileResult)
   }
 }
 
 object AutoTuner extends Logging {
-  val DEFAULT_WORKER_INFO: String = "."
+  // Amount of GPU memory to use per concurrent task in megabytes.
+  // Using a bit less than 8GB here since Dataproc clusters advertise T4s as only having
+  // around 14.75 GB and we want to run with 2 concurrent by default on T4s.
+  val DEF_GPU_MEM_PER_TASK_MB = 7500L
+  // Maximum number of concurrent tasks to run on the GPU
+  val MAX_CONC_GPU_TASKS = 4L
+  // Amount of CPU memory to reserve for system overhead (kernel, buffers, etc.) in megabytes
+  val DEF_SYSTEM_RESERVE_MB: Long = 2 * 1024L
+  // Fraction of the executor JVM heap size that should be additionally reserved
+  // for JVM off-heap overhead (thread stacks, native libraries, etc.)
+  val DEF_HEAP_OVERHEAD_FRACTION = 0.1
+  val MAX_JVM_GCTIME_FRACTION = 0.3
+  // Ideal amount of JVM heap memory to request per CPU core in megabytes
+  val DEF_HEAP_PER_CORE_MB: Long = 2 * 1024L
+  // Maximum amount of pinned memory to use per executor in MB
+  val MAX_PINNED_MEMORY_MB: Long = 4 * 1024L
+  // value in MB
+  val MIN_PARTITION_BYTES_RANGE_MB = 128L
+  // value in MB
+  val MAX_PARTITION_BYTES_RANGE_MB = 256L
+  // value in MB
+  val MAX_PARTITION_BYTES_BOUND_MB: Int = 4 * 1024
+  val MAX_PARTITION_BYTES: String = "512m"
+  val DEF_SHUFFLE_PARTITIONS = "200"
+  val DEF_SHUFFLE_PARTITION_MULTIPLIER: Int = 2
+  // Default CPU cores when missing in cluster properties
+  val DEF_WORKER_CPU_CORES: Int = 16
+  // Default GPU count based on number of cores.
+  // GPU by default is to 1 for each 16 cores
+  val DEF_WORKER_CPU_SORES_PER_GPU = 16
+  val DEFAULT_WORKER_INFO_PATH = "./worker_info.yaml"
   val SUPPORTED_SIZE_UNITS: Seq[String] = Seq("b", "k", "m", "g", "t", "p")
 
-  def loadSystemProperties(fileInput: String): Option[java.util.Map[String, Any]] = {
-    val filePath = new Path(fileInput)
-    val fs = FileSystem.get(filePath.toUri, new Configuration())
-    val yaml = new Yaml()
+  val commentsForMissingProps: mutable.Map[String, String] = mutable.LinkedHashMap[String, String](
+    "spark.executor.memory" ->
+      "'spark.executor.memory' should be set to at least 2GB/core.",
+    "spark.executor.instances" ->
+      "'spark.executor.instances' should be set to (gpuCount * numWorkers).",
+    "spark.task.resource.gpu.amount" ->
+      "'spark.task.resource.gpu.amount' should be set to Max(1, (numCores / gpuCount)).",
+    "spark.rapids.sql.concurrentGpuTasks" ->
+      s"'spark.rapids.sql.concurrentGpuTasks' should be set between [1, $MAX_CONC_GPU_TASKS]",
+    "spark.rapids.memory.pinnedPool.size" ->
+      s"'spark.rapids.memory.pinnedPool.size' should be set up to ${MAX_PINNED_MEMORY_MB}m.",
```

Comment:
like I mention above I would rather see this set to 2GB by default and say user should add to the
spark.executor.memoryOverhead if needed to account for this.



<br class="Apple-interchange-newline">

Reply:
I don't think this was actually changed stll referencing 4G